### DISCUSSION TITLE
Post-"1.0" versioning scheme

### DISCUSSION THREAD

**Discussion by NiklasGustafsson** [xamarin/TorchSharp#361] [General]
Once we decide to move the major version number from 0 to 1, should the major/minor version numbers be independent of PyTorch, or align with them?

A complication with detached versions is that we will always have to explain and document the mapping of TorchSharp versions to LibTorch, which may cause confusion.

A complication with aligned version numbers is that the third number, the patch version, will probably need to be updated more frequently for TorchSharp, as we continue to improve quality and feature set. Thus, it will only realistically be the major/minor versions that will actually align.

**Comment by GeorgeS2019** [comment]
This is feedbacked [here](https://github.com/dotnet/TorchSharp/issues/420).  For new user, it could be confusing currently finding [which version ](https://github.com/dotnet/TorchSharp/blob/main/RELEASENOTES.md)corresponding to the latest LibTorch.  It is incredible that the progress is fast approaching 1.0 TochSharp version!

**Reply by NiklasGustafsson** [reply]
@GeorgeS2019 -- at the top of the RELEASENOTES, it says:

```
NuGet Version 0.95.1
This version integrates with LibTorch 1.10.0.
```

At the top of the README.md, it says:

```
If you are using TorchSharp from NuGet, you should be using a version >= 0.95.1 of TorchSharp, and >= 1.10.0.1 of the libtorch-xxx redistributable packages.
```

What is the confusing part? I want to make sure we resolve it, but it doesn't stand out to me.

**Reply by GeorgeS2019** [reply]
Niklas, for a python or PyTorch user first time coming to TorchSharp, the person is expected to be unfamiliar with LibTorch concept ( there is no need for them to know that unless planning c++ and PyTorch) and where to find libTorch and download it. 

Before getting 1.0, it is normal new user will have question. [SORRY ==> confusing is too strong!! :-)  ]

The Microsoft team has done a great job to make compiling TorchSharp seemless straight forwards, without which, I have failed numerous times in the past each time trying to get into TorchSharp. Making LibTorch easy to integrate to TorchSharp is key to up-take for more TorchSharp users. => This part is done. There is always SLIGHT refinement based on new user feedback.
